funktionale-try/src/test/kotlin/org/funktionale/tries/TryTest.kt

```diff
@@ -54,14 +54,75 @@ class TryTest {
         assertEquals(failure.orElse { Success(5) }.get(), 5)
     }
 
-    @Test fun foreach() {
-        success.foreach { assertEquals(10, it) }
-        failure.foreach { fail() }
+    @Test fun foreach_success_sideEffect() {
+        var wasInside = false
+        success.foreach { wasInside = true }
+        assertTrue(wasInside)
     }
 
-    @Test fun onEach() {
-        assertEquals(success.onEach { assertEquals(10, it) }.get(), 10)
-        assertTrue(failure.onEach { fail() }.isFailure())
+    @Test fun foreach_failure_sideEffect() {
+        var wasInside = false
+        failure.foreach { wasInside = true }
+        assertFalse(wasInside)
+    }
+
+    @Test fun foreach_success_throwInside() {
+        try {
+            success.foreach { throw RuntimeException("thrown inside") }
+        } catch (e: Exception) {
+            assertTrue(e is RuntimeException)
+            assertEquals(e.message, "thrown inside")
+        }
+    }
+
+    @Test fun foreach_failure_throwInside() {
+        try {
+            failure.foreach { throw RuntimeException("thrown inside") }
+        } catch (e: Exception) {
+            assertTrue(e is NumberFormatException)
+        }
+    }
+
+    @Test fun onEach_success_sideEffect() {
+        var wasInside = false
+        success.onEach { wasInside = true }
+        assertTrue(wasInside)
+    }
+
+    @Test fun onEach_failure_sideEffect() {
+        var wasInside = false
+        failure.onEach { wasInside = true }
+        assertFalse(wasInside)
+    }
+
+    @Test fun onEach_success_throwInside() {
+        try {
+            success.onEach { throw RuntimeException("thrown inside") }.get()
+        } catch (e: Exception) {
+            assertTrue(e is RuntimeException)
+            assertEquals(e.message, "thrown inside")
+        }
+    }
+
+    @Test fun onEach_failure_throwInside() {
+        try {
```

Comment:
Moreover thanks to your comment, @jkpl , I found a false positive in my test for `foreach_failure_throwInside`: it was assertion an exception inside `catch` and `foreach` is not designed to throw an exception. Thanks 👍